src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -389,3 +490,378 @@ def names(self, names):
+
+            subprocess_cmd = kubectl_prior + ["delete", "--all", "apd", "-n", "aks-periscope", "--ignore-not-found"]
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            subprocess_cmd = kubectl_prior + ["delete", "CustomResourceDefinition", "diagnostics.aks-periscope.azure.github.com", "--ignore-not-found"]
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            print()
+            print(f"{colorama.Fore.GREEN}Deploying diagnostic container on the K8s cluster...")
+            subprocess_cmd = kubectl_prior + ["apply", "-f", temp_yaml_path, "-n", "aks-periscope"]
+            subprocess.check_output(subprocess_cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as err:
+            raise CLIInternalError(err.output)
+    finally:
+        os.remove(temp_yaml_path)
+
+    print()
+    # log_storage_account_url = f"https://{storage_account_name}.blob.core.windows.net/"
+
+    print(f'{colorama.Fore.GREEN}Your logs are being uploaded to storage account {format_bright(storage_account_name)}...')
+
+    print()
+    print(f'You can download Azure Storage Explorer here '
+          f'{format_hyperlink("https://azure.microsoft.com/en-us/features/storage-explorer/")}'
+          f' to check the logs by accessing the storage account {storage_account_name}.')
+    # f' to check the logs by adding the storage account using the following URL:')
+    # print(f'{format_hyperlink(log_storage_account_url)}')
+
+    print()
+    if not prompt_y_n('Do you want to see analysis results now?', default="n"):
+        print(f"You can rerun 'az connectedk8s troubleshoot -g {resource_group_name} -n {name}' "
+              f"anytime to check the analysis results.")
+    else:
+        display_diagnostics_report(kubectl_prior)
+
+    try:
+        print("Deleting aks-periscope resources from cluster ...")
+
+        subprocess_cmd = kubectl_prior + ["delete", "serviceaccount,configmap,daemonset,secret", "--all", "-n", "aks-periscope", "--ignore-not-found"]
+        subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+        subprocess_cmd = kubectl_prior + ["delete", "ClusterRoleBinding", "aks-periscope-role-binding", "--ignore-not-found"]
+        subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+        subprocess_cmd = kubectl_prior + ["delete", "ClusterRole", "aks-periscope-role", "--ignore-not-found"]
+        subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+        subprocess_cmd = kubectl_prior + ["delete", "--all", "apd", "-n", "aks-periscope", "--ignore-not-found"]
+        subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+        subprocess_cmd = kubectl_prior + ["delete", "CustomResourceDefinition", "diagnostics.aks-periscope.azure.github.com", "--ignore-not-found"]
+        subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as err:
+        raise CLIInternalError(err.output)
```

Comment:
Don't raise **CLIInternalError** here(CLI will exit from here itself in this case), rather just use Fplain Exception, so that this can be caught by the outer try-catch.

and Maybe add some descriptive log instead on just err.output

something like - **_raise Exception("Error occurred while deleting the aks-periscope logs. Error: {}".format(err.output))_**